### DISCUSSION TITLE
Ein Victron mppt via ve.direct an zwei ESPs anschließen

### DISCUSSION THREAD

**Discussion by openKimchi** [hoylabs/OpenDTU-OnBattery#1394] [General]
Moinsen,
vermutlich geht es nicht aber ich frage trotzdem mal: Ist es möglich via ve.direct ein Victron mppt an zwei verschiedene ESPs anzuschließen. Also gleichzeitig an ein ESP32 S3 mit OpenDTU onBattery und ein ESP8266 mit Victron2mqtt drauf?

**Comment by SW-Niko** [comment]
Hmmm... eigentlich ist der VE.Direct kein Bus sondern nur eine Punkt zu Punkt Verbindung.
Es sollte aber trotzdem klappen, wenn du dich auf den Empfang der Daten beschränkst. (Tx in beiden ESP auf -1)
Und beide VE.Direct Verbindungen zu den ESPs über ADUMs trennst.
Dann kannst du jeweils die Leitungen VCC, GND und Transmit am MPPT verbinden. Verstanden? Oder benötigst du einen Schaltplan?

**Comment by openKimchi** [comment]
Danke für deine Hilfe!
Einen Schaltplan bräuchte ich wohl nicht, wenn ich es richtig verstanden habe aus den einfachen Kabeln jeweils y-Kabel machen.
Was ich aber zwingend bräuchte, wäre TX, zumindest an einem der beiden ESPs.

**Reply by SW-Niko** [reply]
Ein ESP mit Tx/Rx und einer nur mit Rx sollte elektrisch kein Problem sein. **Galvanische Trennung für beide ESP nicht vergessen!**
Ob die Software mitspielt kann ich dir nicht versprechen. OpenDTU-OnBattery prüft ob Antworten zu den Kommandos passen.
Probier es aus. Ich denke du hast gute Karten das es klappt.

**Comment by openKimchi** [comment]
vielen Dank! Ist die galvanische Trennung hier besonders wichtig? Bisher bin ich immer gut ohne ausgekommen.
Könntest du mir dafür doch Mal einen Schaltplan aufzeichnen? Ich habe das zweimal mit so einem ADUM probiert. Hat nie geklappt. Entweder sind die Dinger sehr empfindlich beim Löten, oder ich habe sie falsch angeschlossen. Rumliegen habe ich allerdings noch einige...

**Reply by schlimmchen** [reply]
Was fehlt dir denn in der [Doku](https://opendtu-onbattery.net/hardware/vedirect/#adum1201-safest-most-preferred)?

**Reply by SW-Niko** [reply]
>Ist die galvanische Trennung hier besonders wichtig?

Kurz gesagt: Ja. 

Schon möglich das du darauf verzichten kannst. Wenn du dich gut mit Elektrotechnik auskennst und klar Begründen kannst, warum du nicht galvanisch trennst. Ansonsten würde ich auf Nummer sicher gehen.

Schau dir mal hier #1374 das PDF an.

>Könntest du mir dafür doch Mal einen Schaltplan aufzeichnen?

Klar, aber du must dich noch etwas gedulden. (Momentan bin ich noch mit meiner eigenen Anlage etwas im Umbau-Fieber)

**Reply by openKimchi** [reply]
ich gedulde mich...

**Comment by openKimchi** [comment]
ja danke, danach hatte ich es damals gemacht. Vielleicht war mein Lötkolben ja zu heiß...sind die Teile so empfindlich?

**Reply by SW-Niko** [reply]
Nö, eigentlich nicht.

**Reply by SW-Niko** [reply]
Bitte schön ....

![grafik](https://github.com/user-attachments/assets/6164d1ba-a400-4705-b2bf-2ffa1e7792bc)

**Reply by openKimchi** [reply]
Herzlichen Dank!

**Reply by SW-Niko** [reply]
Hallo @openKimchi ,
kannst du noch die Frage von schlimmchen beantworten. Dann können wir die Dokumentation um die fehlenden Informationen erweitern. Danke.

**Reply by openKimchi** [reply]
ich hatte gedacht, die Frage sei ironisch gemeint. Aber wenn du schon fragst, habe ich persönlich deine obige Schaltung besser/schneller verstanden, als die in der Hilfe. Liegt aber vermutlich auch an meiner Faulheit, mich mit dem englischen auseinander zu setzen. Unter "Voltage Domain" kann ich mir auf Anhieb nicht so viel vorstellen. Wie wäre es, wenn ihr deinen obigen Schaltplan als Beispiel dazu packt? Am coolsten wäre es natürlich, wenn es die gesamte Doku auch auf Deutsch gäbe, aber das wäre wohl etwas zuviel verlangt.

**Reply by openKimchi** [reply]
btw. hat mich Schlimmchen aber auch darauf hingewiesen, dass es in zukünftigen Versionen möglich sein soll, die Daten des Ladereglers per mqtt an openDTU o.B. zu senden, dann wäre die Thematik mit ein Laderegler via ve.direct an 2 ESPs für mich eh obsolet. Das nur so nebenbei, falls noch andere auf die Idee kommen sollten, das zu versuchen.

**Reply by SW-Niko** [reply]
Hallo @schlimmchen ,
ich habe den Plan aufgehübscht und an den Text in der Hilfe angepasst.
Vielleicht hilft es dem einen oder anderen, wenn wir zusätzlich zur Beschreibung noch eine visuelle Unterstützung einfügen. 🤔 

![grafik](https://github.com/user-attachments/assets/ef936ca8-636c-4705-8bb4-711acff90632)

**Reply by schlimmchen** [reply]
Sehr cool! Ich weiß, dass in einer solchen Grafik, mag sie auch "einfach" aussehen, viel Arbeit steckt, erst recht wenn sie so schön sauber und aufgeräumt ist. Vielen Dank!

Das packen wir dann in die [Doku](https://opendtu-onbattery.net/hardware/vedirect/#adum1201-safest-most-preferred), oder?

Ist dein Upload schon in Originalauflösung oder müssen wir einen Weg finden das Ding hochauflösend zu übergeben?

Es fehlt ein weißer Rand. Du vergisst, dass nicht jeder sich vor dem Bildschirm versucht zu sonnen. Die dunklen Umrandungen der Kästen links und rechts gehen nahtlos in einen dunklen Hintergrund über.

![image](https://github.com/user-attachments/assets/7cefab43-5b12-4ad3-a3cb-69a13d06e783)

**Reply by schlimmchen** [reply]
> ich hatte gedacht, die Frage sei ironisch gemeint.

@openKimchi Ne, das war ernst gemeint. Normalerweise schreibe ich "Wie müsste man die Doku verbessern damit du das verstanden hättest" und so konstruktiv ist es auch gemeint.

Eine zusätzliche Doku in Deutsch werde ich definitiv nicht pflegen. Dein Browser sollte dir, spätestens mit Hilfe eines Plugins, anbieten, die Webseite zu übersetzen, wenn du das brauchst. Die Ergebnisse dieser automatischen Übersetzung sind sicherlich hinreichend gut.

**Reply by SW-Niko** [reply]
Bitte schön

![grafik](https://github.com/user-attachments/assets/d21561c2-3f08-4c1a-ae96-d0baa788823c)

Du kannst aber auch die LibreOffice-Draw Datei haben. 
Ich habe noch 2 optionale Kondensatoren eingefügt. Die ADuM benötigen sehr wenig Strom und die Frequenz der VE.Direct ist auch sehr gemächlich. Wenn jemand Leitungen über 1m hat und Probleme mit der Signalintegrität, dann sind die Kondensatoren hilfreich.

**Comment by Ralf-64** [comment]
Hallo Allerseits,
 @openKimchi: 2 Esp's am Victron MPPT funktioniert. Habe hier einen ESP mit "Victron2MQTT" und ein einem Fusion Board mit Open DTU on Batterie in Betrieb.
Auf dem Fusion Board ist ein Can/ Iso Board installiert (Galvanische Trennung, Pegelwandler). Das NodeMcu ist direkt angeschlossen und liefert auch TX Signal um den MPPT Regler Ein/ Ausschalten zu können.
Einen ADuM WandlerTrenner vor das NodeMcu zu schalten wäre sicherlich die elegantere Lösung....

LG Ralf

**Comment by Ralf-64** [comment]
![20240928_173315](https://github.com/user-attachments/assets/d7a524ac-a3db-4b00-8385-4ec0d125da85)

In der Kartons mit den Displays sind Fusion Boards, vorne auf den MPTT's die NodeMcu's.... 
Ja, das mache ich noch ordentlich ;-))

**Reply by openKimchi** [reply]
OK, danke für die Info!
Ich habe da nochmal eine Verständnisfrage: Wofür wird die Power Leitung kommend vom Victron Gerät eigentlich am ESP32 mit openDTU o.B. gebraucht? Der ESP8266 mit Victron2MQTT nutzt die nämlich gar nicht. Siehe auch [hier](https://github.com/softwarecrash/Victron2MQTT/wiki/Wiring-Diagram)
Benötigt der ADUM das? Und wenn ja, könnte man dann nicht auch den 5V Anschluss am ESP32 benutzen?

**Reply by schlimmchen** [reply]
> Wofür wird die Power Leitung kommend vom Victron Gerät eigentlich am ESP32 mit openDTU o.B. gebraucht?

Wird sie nicht und sollte auch nicht verbunden werden.

Wenn du allerdings einen ADUM1201 oder wenn du Level-Shifter einsetzt, dann musst du natürlich die Versorungsspannung vom VE.Direct Gerät anschließen an den ADUM1201 bzw. an den LEvel-Shifter.

**Reply by Sorbat** [reply]
> ![20240928_173315](https://github.com/user-attachments/assets/d7a524ac-a3db-4b00-8385-4ec0d125da85)
> 
> 
> 
> In der Kartons mit den Displays sind Fusion Boards, vorne auf den MPTT's die NodeMcu's.... 
> 
> Ja, das mache ich noch ordentlich ;-))

Hallo, ich hatte mal ein Gehäuse für das Fusion für eine Hutschienenmontage erstellt, das könnte bei dir ganz gut passen. 
Hast du einen Drucker und hättest du Interesse daran? Dann würde ich mal schauen, wie ich dir die Datei zukommen lassen kann

**Comment by Ralf-64** [comment]
Hallo
@openKimchi :
Weiß jetzt nicht wie weit ich ausholen soll...
Es geht hierbei um die galvanische Trennung der Can/ISO Eingänge des Fusion Board. Sieh dir die Zeichnung von SW-Niko (tolle Arbeit !!) weiter oben an.
Der ADuM Baustein hat 2 "Anschlußseiten". Pin 1 bis 4 und 5 bis 8. Jede Seite benötigt eine galanisch getrennte Versorgungsspannung (Plus u. Masse) um diese Entkopplung zu erreichen.
Falls du den MPPT direkt an den ESP8266 hängen magst, kannst du durchaus auch diesen Plus für den Can/ISO Shield am Fusion Board nehmen, vorausgesetzt er hängt nicht auf dem gleichen Plus wie das Fusion Board. Der Plus vom MPPT ist potentialfrei.
Besser wäre es die Verbindung zwischen dem ESP8266 und dem MPPT mit einem Adum zu entkoppeln u. die Pegel auf korrekte 3,3Volt zu wandeln.
Werde diese Änderung bei Gelegenheit an meiner Anlage nachrüsten.

LG Ralf

**Reply by openKimchi** [reply]
Ich bin zu blöd. Habe nochmal 2 ADUMs zusammengelötet aber die Dinger funktionieren bei mir einfach nicht. Beim ESP8266 bekomme ich ohne ADUM das Signal, schalte ich sie dazwischen, passiert nichts.

**Reply by schlimmchen** [reply]
Okay, wir wissen jetzt, dass du zu blöd bist (deine Worte). Und jetzt? Wie sollen wir dir helfen? Magst du mal genau dokumentieren, wie du es verlötet hast und jeweils angeklemmt hast? Dabei wirst du dann deinen Aufbau selbst nochmal reflektieren können und wir können dann womöglich noch einen Tipp geben. Allerdings ist [Nikos Diagramm](https://github.com/hoylabs/OpenDTU-OnBattery/discussions/1394#discussioncomment-11272412) eigentlich schon das beste, was man remote helfen kann...

**Reply by openKimchi** [reply]
Ich habe nur die PIN Leisten eingelötet und das ganze dann auf ein Breadboard zum Testen gesteckt. Viel kann man da ja dann nun nicht falsch stecken. 

![image](https://github.com/user-attachments/assets/686ccd39-839b-48e8-9e7a-086a5e490325)

**Reply by SW-Niko** [reply]
Leider sehe ich nicht alles. Die linke schwarze Leitung ist die Tx vom MPPT und die rechte gelbe geht auf den Rx vom ESP32?
Du kannst auch mal den ADuM entfernen und 2 zusätzliche Brücken stecken

![grafik](https://github.com/user-attachments/assets/d693bfd5-82ea-46f2-958e-cf4ee99958de)

Wenn das funktioniert dann ist der ADuM defekt. Wenn das auch nicht funktioniert dann ist deine Verkabelung oder Konfiguration falsch.

Ich kann es nicht sehen, aber du hast den ADuM richtig eingelötet und nicht um 180° gedreht?
Und es muss ein ADuM1201 sein. Ein ADuM1200 geht nicht.

**Reply by Sorbat** [reply]
Er schrieb doch, dass er nur die Pinleisten angelötet hat. Das hört sich an, als würde er damit die 2,54er leisten meinen und die Adumen bereits bestückt sein 

Und scheinbar gibt es diese lila Dinger auch genau so zu kaufen ![image](https://github.com/user-attachments/assets/41029410-be5b-4e0b-940f-c74b0f5bed17)

**Reply by SW-Niko** [reply]
Aha, Ok ... dann kann ich schon mal einige Fehlerquellen streichen. 👍

**Reply by openKimchi** [reply]
Ich habe mittlerweile bestimmt alle möglichen Verbindungsvarianten durch und mittlerweile bekomme ich auch kein Signal mehr, wenn ich direkt verbinde. Vermutlich habe ich den ESP abgeschossen oder er müsste neu geflashed werden. Mir ist aber auch jetzt die Lust vergangen. Vielleicht wird es ja doch irgendwann die Möglichkeit geben aus openDTU o.B. heraus den RX Port anzusprechen oder die Daten vom Laderegler per MQTT in openDTU o.B. zu laden. Momentan is ja wohl eh eher mau mit Überschussladung.
Danke trotzdem für eure Mühe und Hilfe!!!

**Reply by SW-Niko** [reply]
Schade das wir deinen Aufbau nicht zum laufen gebracht haben.
Aber ..... Du hast mich motiviert einen Schaltplan zu zeichnen und der hilft vielleicht andern beim Aufbau. 👍

**Reply by openKimchi** [reply]
Kein Problem, ich habe mich echt gefreut über die Hilfe. Und dein Plan ist wirklich richtig gut. Und wie gesagt, das Problem wird sich durch zukünftige Innovationen von alleine lösen. Bis die Sonne mal wieder länger scheint ist es ja noch etwas hin.

**Comment by Ralf-64** [comment]
Hallo Sorbat,das mit Hutschienen Montage hört sich interessant an...3D Drucker ist vorhanden. Mit was konstruierst du ?? Bin im Moment bei Freecad 21.x .Siehst du meine E-Mail Adresse ? Ansonsten ***@***.*** .. (EDIT(schlimmchen): von GitHub automatisch unkenntlich gemacht)
Danke dir !!!
LG Ralf

**Reply by Sorbat** [reply]
Das mit der email Adresse funktioniert nicht so ganz 😂 
Ich nutze Fusion. Ich versuche heute Abend mal dran zu denken, die Teile zu exportieren und auf ein googledrive oder sowas zu laden. 
Eigentlich sollte das so passen, sodass da nichts mehr angepasst werden muss 

Da ich gerade neben der dtu sitze 😅
Hier mal ein Foto 
![image](https://github.com/user-attachments/assets/c04d4d45-c915-4ccb-9afc-8a1d33ea5b69)

**Reply by schlimmchen** [reply]
Pack die Daten in eine ZIP und lade sie als Anhang an einem Kommentar hier einfach hoch :+1: Darf ich die dann auch über die [Doku](https://opendtu-onbattery.net/3rd_party/cases/) anpreisen? Falls ja, was wäre die Kurbeschreibung? Was ist das für ein(e) Display(größe)?

**Reply by Sorbat** [reply]
Kannst du Dateien als Direktdownload da reinstellen? Oder soll ich das dann auf Thingiverse hochladen?

**Reply by schlimmchen** [reply]
Das ist eine gute Frage, die ich mir auch schon gestellt habe, aber verschieben wollte. Im äußersten Notfall kann man das auch embedden, aber vermutlich kann man es auch einfach in der Doku als Datei hinterlegen. Dennoch wäre vermutlich ideal, wenn es gleich auf thingiverse ist. Aber ich mute dir nicht zu, dass du deshalb dort einen Account erstellen musst. Es geht auch anders.

**Reply by Sorbat** [reply]
Ne ich hab dort einen Account. 
Ich hatte das bisher nicht veröffentlicht, weil ich zum einen zu faul dazu war (und ein anderes Projekt hatte 😄), zum anderen ich aber auch kein Interesse an dieser Version verzeichnen konnte. 
Außerdem wollte ich eigentlich diesen Verriegelungsmechanismus nicht unbedingt puplishen, weil ich da echt lange dran saß und hinterher sowas dann bei Kleinanzeigen oder so verkauft wird 😄 aber wenn es niemand nutzen kann, bringt es auch nicht viel.
Ich lad es gleich mal hoch und schick dir dann einen pr

**Reply by Sorbat** [reply]
@Ralf-64 schau mal hier https://www.thingiverse.com/thing:6839520
das müsste soweit alles passen. Bei diesem Displaycover für das 1,54" gibt es 3 Versionen, eine normale, eine mit dem onBattery Batteriesymbol für Mulitcolordruck und eine mit dem Symbol als negativ. Für das 1,3" Zoll hab ich das mit dem Symbol nicht gemacht, weil mir persönlich das Display zu klein ist und ich keinen wirklichen Sinn für dieses Display sehe - entweder das 1,54" auf dem man auch was erkennt, oder halt kein Display.
Manche Teile müsstest du dir ggfs. noch im Slicer umdrehen, sodass jeweils die flache Seite auf dem Druckbett liegt.
Da ich nur HMS Wechselrichter habe, hab ich dort Aussparungen für nur 2 Antennen gemacht. Es gibt aber unter dem PCB eine Aussparung für eine 3. (Klebe-)antenne, sollte man diese benötigen. wenn du eine richtige 3. Antenne einbauen willst oder sonst eine Änderung brauchst, sag bescheid, dann passe ich das an.

**Reply by schlimmchen** [reply]
> hinterher sowas dann bei Kleinanzeigen oder so verkauft wird

Arschlöcher gibt es überall :man_shrugging: 

> Ich lad es gleich mal hoch und schick dir dann einen pr

Mega! :heart:
